OSDOCS-3889 add NLB to AWS networking config (#51460)
openshift/openshift-docs: modules/installation-aws-config-yaml.adoc

[id="installation-aws-config-yaml_{context}"]
= Sample customized install-config.yaml file for AWS

> You can customize the installation configuration file (`install-config.yaml`) to specify more details about your {product-title} cluster's platform or modify the values of the required
> parameters.

Reviewer: ```suggestion
You can customize the installation configuration file (`install-config.yaml`) to specify more details about your {product-title} cluster's platform or modify the values of the required parameters.
```

  name: master
  platform:
    aws:

>       lbType: nlb

Reviewer 2: It can either be "NLB" or "Classic". By default if not set it will be "Classic". cc @Miciah

>       lbType: nlb <12>

Reviewer 2: NLB

====
For production {product-title} clusters on which you want to perform installation debugging or disaster recovery, specify an SSH key that your `ssh-agent` process uses.
====
ifndef::vpc,restricted[]

> <12> You can provide either the `NLB` or `Classic` load balancer type in this parameter. If you do not specify a value, the installation program defaults to `Classic`.

Reviewer 3: Is there a particular reason you're adding this as callout number 12?

Usually we'd want the callouts to go in order, so where this is now it should be number 6. Unless there's a reason it needs to be like this due to the crazy ifdefs?

Author: Yes, it's the crazy ifdefs. I couldn't change them all.

Reviewer 3: It's probably something we should fix at some point, but honestly I'd rather it be out of order than throw things off due to how complex this is. Let's merge and put it on the back burner to look at another day!

====
For production {product-title} clusters on which you want to perform installation debugging or disaster recovery, specify an SSH key that your `ssh-agent` process uses.
====
ifndef::vpc,restricted[]

> <12> You can provide either the `NLB` or `Classic` load balancer type in this parameter. The default load balancer type is Classic if no value is provided.

Reviewer: ```suggestion
<12> You can provide either the `NLB` or `Classic` load balancer type in this parameter. If you do not specify a value, the installation program defaults to `Classic`.
```
So you can refer to the specific object reference and use backtics (`Classic`) or refer to the general object reference (classic load balancer type).

See the [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references) for help in this area. I still have a hard time with this concept and refer to this section of the guidelines often. Many writers default to using literals when they are not sure what to do.

====
For production {product-title} clusters on which you want to perform installation debugging or disaster recovery, specify an SSH key that your `ssh-agent` process uses.
====
ifndef::vpc,restricted[]

> <12> You can provide another load balancer type, such as `nlb`, in this parameter. The load balancer type is Classic by default.

Reviewer 2: You can specify NLB or Classic. However default lbType is Classic if no value is provided.
